### ISSUE TITLE
Issue with scala211 and Hydrogen

### ISSUE THREAD

**Issue by sbromberger** [alexarchambault/jupyter-scala#38]
Hi,

First, thanks for this package. It looks very cool and it works well using the ipython repl.

However, it's not working under Hydrogen (an Atom editor plugin). See https://github.com/willwhitney/hydrogen/issues/121#issuecomment-148939094 and following comments for more information, and I'm happy to post data from my installation here as well. I'm just not sure what you need, or even where the issue is.

In short: the kernel starts, but then just hangs (no output; the execution spinner keeps spinning).

It'd be great to be able to use scala with Hydrogen!

**Comment by alexarchambault** [comment]
Thanks for having tried that!

I just gave it a try, and think the problem originates from hydrogen giving as IDs to its messages things that are not UUIDs. Getting this in the log of the kernel that is launched:
```
015-12-12 17:19:39 DEBUG [pool-2-thread-3] j.k.s.z.ZMQStreams$ [ZMQStreams.scala:147] Read message (List(WrappedArray(100, 101, 97, 108, 101, 114, 115, 99, 97, 108, 97, 53, 54, 56, 57, 51)),4cf0f89063678f7d00fe3bf8e2e6b8d66ee2205d6936ec78b85516cda43e68c8,{"msg_id":"execute_d248e319-721d-4273-b1e4-ac6518614af0","username":"","session":"00000000-0000-0000-0000-000000000000","msg_type":"execute_request","version":"5.0"},{},{},{"code":"2","silent":false,"store_history":true,"user_expressions":{},"allow_stdin":false}) on Requests
2015-12-12 17:19:39 DEBUG [pool-2-thread-3] j.k.s.InterpreterServer$ [InterpreterServer.scala:97] Error while handling message: Decoding message: Invalid UUID: execute_d248e319-721d-4273-b1e4-ac6518614af0: [--\(msg_id)]
```

This should be fixed in [jupyter-kernel](https://github.com/alexarchambault/jupyter-kernel) I think, that deals with this kind of things.

I'll have time to fix that as soon as I give some love to jupyter-scala again (which should be after a release of [coursier](https://github.com/alexarchambault/coursier)).

**Comment by rgbkrk** [comment]
Reiteration from #146:

> I'm on a Mac with the most recent version of Hydrogen, coursier, and jupyter-scala. It seems to be working well.

![screen shot 2017-04-24 at 10 19 15 pm](https://cloud.githubusercontent.com/assets/836375/25369925/2d3f8ca4-293c-11e7-89ec-c3bba225096d.png)

> One extra thing I had to do was install the `language-scala` package.

**Comment by rgbkrk** [comment]
> I just gave it a try too, and think the problem originates from hydrogen giving as IDs to its messages things that are not UUIDs.

Hydrogen _should_ be using uuidv4 now, so hopefully that's not happening any more.

**Comment by alexarchambault** [comment]
Thanks @rgbkrk!

Closing this one, as hydrogen should be using standard uuids *and* jupyter-scala should be fine with non-UUIDs message IDs too (since `0.4.0` at least).
